### PR TITLE
Add commonly used filter functions.

### PR DESCRIPTION
**Pull request by rainwoodman**
This introduces a new module, nbodykit.filter.

### CODE REVIEW

File: nbodykit/filter.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    def filter(self, k, v):
+        r = self.r
+        k = sum(ki ** 2 for ki in k) ** 0.5
+        kr = k * r
+        w = 3 * (numpy.sin(kr) / kr **3 - numpy.cos(kr) / kr ** 2)
+        w[k == 0] = 1.0
+        return w * v
+
+class Gaussian(MeshFilter):
```
Comment (nickhand) [inline nbodykit/filter.py]: FourierSpaceGaussian?

File: nbodykit/filter.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+from nbodykit.base.mesh import MeshFilter
+
+import numpy
+
+class TopHat(MeshFilter):
```
Comment (nickhand) [inline nbodykit/filter.py]: FourierSpaceTopHat? KSpaceTopHat? ...we should probably differentiate from the real-space equivalent

**Comment by rainwoodman** [comment]
oops Python 2.x

File: nbodykit/tests/test_filter.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
+from nbodykit.lab import *
+from nbodykit import setup_logging
+from runtests.mpi import MPITest
+
+import pytest
+
+# debug logging
+setup_logging("debug")
+
+from nbodykit.filter import TopHat, Gaussian
```
Comment (nickhand) [inline nbodykit/tests/test_filter.py]: Shall we import filter in nbodykit.lab?

File: nbodykit/filter.py (hunk 2 above)
Comment (rainwoodman) [reply nbodykit/filter.py]: Not sure yet. Probably filter.fourier.TopHat?

Comment (rainwoodman) [reply nbodykit/filter.py]: We can always convert the class argument to an argument to support configuration space convolutions.

**Pull request merged**